.github/workflows/testing.yml

```diff
@@ -3,6 +3,10 @@ name: Ensure PR
 on:
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
```

Comment:
Should we define it at the job level?

Reply:
I think it makes more sense to run at workflow level since we need to rerun all jobs against the latest commit and cancel any running jobs, besides, it's much simpler and easier to maintain.